### PR TITLE
Close #6720: Update current URL in onLocationChange

### LINKED ISSUE #6720: Session should update current URL in onLocationChange
Currently `GeckoEngineSession` is updating the current URL in `onPageStart`.  This does not correct because the page load might fail and the URL will be rolled back.  `onLocationChange` is when the URL change is committed.  Update current URL in `onLocationChange`

### PR DESCRIPTION
**Pull request by rocketsroger**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

File: components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt [hunk 1]
```diff
@@ -568,28 +568,51 @@ class GeckoEngineSessionTest {
...
         progressDelegate.value.onPageStart(geckoSession, "https://www.mozilla.org")
-        assertEquals("https://www.mozilla.org", engineSession.currentUrl)
+        assertNull(engineSession.currentUrl)
 
         progressDelegate.value.onPageStart(geckoSession, "https://www.firefox.com")
+        assertNull(engineSession.currentUrl)
+    }
+
+    @Test
+    fun `keeps track of current url via onPageStart events`() {
```
Comment (csadilek) [inline components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt:586]: flyby: nit: this should be `onLocationChange` :)

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt [hunk 2]
```diff
@@ -356,6 +356,8 @@ class GeckoEngineSession(
             if (initialLoad && url == ABOUT_BLANK) {
                 return
             }
+
+            currentUrl = url
```
Comment (csadilek) [inline components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:360]: Looks like we only use the `currentUrl` in `onLocationChange` and `onTitleChange`. Did you verify this still works correctly (in the right order). :)

File: components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt (hunk 1 above)
Comment (rocketsroger) [reply components/browser/engine-gecko-nightly/src/test/java/mozilla/components/browser/engine/gecko/GeckoEngineSessionTest.kt:586]: oops, thanks for catching this. 👍  I'll fix

File: components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt (hunk 2 above)
Comment (rocketsroger) [reply components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt:360]: Didn't think about the order.  Just tested and they are in the right order.
Before change: `onPageStart` -> `onTitleChange`
After change: `onLocationChange` -> `onTitleChange`

**Comment by rocketsroger** [comment]
bors r=csadilek

**Pull request merged**